src/multihash.rs

```diff
@@ -30,12 +31,23 @@ pub trait MultihashDigest: Clone + Debug + Eq + Send + Sync + 'static {
         Self: Sized;
 
     /// Parses a multihash from a bytes.
+    ///
+    /// You need to make sure the passed in bytes have the correct length. The digest length
+    /// needs to match the `size` value of the multihash.
     #[cfg(feature = "std")]
     fn from_bytes(mut bytes: &[u8]) -> Result<Self, Error>
     where
         Self: Sized,
     {
-        Self::read(&mut bytes)
+        let result = Self::read(&mut bytes)?;
+        // There were more bytes supplied than read
+        if !bytes.is_empty() {
+            Err(Error::InvalidSize(bytes.len().try_into().expect(
+                "Currently the maximum size is 255, therefore always fits into usize",
+            )))
+        } else {
```

Comment:
nitpick: I'd go with 
```rust
if { return Err(); }
Ok(result)
```
as I think it's nicer to read

Reply:
Sure. Writing it like this comes from my Erlang past.

Reply:
I agree that it's an early return and not an "either or", hence doing it with the return makes sense.